docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,458 @@
+    var.kibana.host: "localhost:5601"
+    var.input.azure_event_hubs.threads: 8
+    var.input.azure_event_hubs.decorate_events: true
+    var.input.azure_event_hubs.consumer_group: "logstash"
+    var.input.azure_event_hubs.storage_connection: "DefaultEndpointsProtocol=https;AccountName=example...."
+    var.input.azure_event_hubs.event_hub_connections:
+      - "Endpoint=sb://example1...EntityPath=insights-logs-errors"
+      - "Endpoint=sb://example2...EntityPath=insights-metrics-pt1m"
+-----
+
+The basic configuration requires the `var.input.azure_event_hubs.` prefix
+before a configuration option. 
+Notice the notation for the `threads` option. 
+
+[[adv-config-sample]]
+===== Advanced configuration sample
+
+Advanced configuration in the `logstash.yml` file supports Event Hub specific options. 
+Advanced configuration is not necessary or recommended for most use cases. Use
+it only if it is required for your deployment scenario. 
+
+You must define the `header` array with `name` in the first position. You can
+define other options in any order. The per Event Hub configuration takes
+precedence. Any values not defined per Event Hub use the global config value.  
+
+In this example 'consumer_group' will be applied to each of the configured Event
+Hubs. Note that 'decorate_events' is defined in both the 'global' and per Event Hub
+configuration. The per Event Hub configuration takes precedence, and the
+global configuration is effectively ignored.
+
+["source","shell",subs="attributes"]
+-----
+modules:
+  - name: azure
+    var.elasticsearch.hosts: "localhost:9200"
+    var.kibana.host: "localhost:5601"
+    var.input.azure_event_hubs.threads: 8
+    var.input.azure_event_hubs.decorate_events: true
+    var.input.azure_event_hubs.consumer_group: logstash
+    var.input.azure_event_hubs.event_hubs:
+      - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
+      - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
+      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
+      - ["insights-logs-errors",      "Endpoint=sb://example3...", "DefaultEndpointsProtocol=https;AccountName=example3....", "TAIL",             "false"]
+      - ["insights-operational-logs", "Endpoint=sb://example4...", "DefaultEndpointsProtocol=https;AccountName=example4....", "HEAD",             "true"]
+-----
+
+The advanced configuration doesn't require a prefix before a per Event Hub
+configuration option. Notice the notation for the `decorate_events` option. 
+
+[[azure_best_practices]]
+===== Best practices
+
+Here are some guidelines to help you avoid data conflicts that can cause lost
+events.
+
+* **Create a {ls} consumer group.** 
```

Comment:
I curated them.  I don't think the existing suggestions here will ever go out of fashion since it is at the core of how the event hubs work. However, there may be new (or existing) best practices not represented here. 

The only best practice I found from MS is https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#best-practices and most of their documentation is standard howto, configuration, pricing, etc. Not much consumer guidance.

Reply:
Got it, thanks for the insight.